scheduler/src/cook/tools.clj

```diff
@@ -916,27 +916,50 @@
        first
        field))
 
-(defn filter-based-on-quota
+(defn get-quota-for-pool
+  "Given a pool name, determine the quota for that pool."
+  [quotas effective-pool-name]
+  (match-based-on-pool-name quotas effective-pool-name :quota))
```

Comment:
Why `effective-pool-name` instead of just `pool-name`?

Reply:
Because it includes the mapping from nil to "no-pool".

Reply:
Isn't that true for almost every variable named `pool-name` in the codebase?

Reply:
No. Some of them are the actual pool name based on the user. Some are with nil mapped to "no-pool" (and I hate that inconsistency).